lib/doi_search.rb

```diff
@@ -12,17 +14,30 @@ def self.search(doi)
       sw_hits = ScienceWireClient.new.get_pub_by_doi(doi)
       return sw_hits unless sw_hits.empty?
     end
-    results = []
-    if Settings.WOS.enabled
-      results += WebOfScience.queries.search_by_doi(doi).map { |rec| add_citation(rec.pub_hash) }
-      full_match = normalized_dois(results)[normalized_doi(doi)]
-      return [full_match] if full_match
+    results = web_of_science(doi)
+    if results.present?
+      name = doi_name(doi)
+      match = results.find { |pub_hash| doi_name(pub_hash[:doi]) == name }
+      return [match] if match.present?
     end
     # Everything else is non-authoritative
     results.unshift(pub.pub_hash) if pub
     results
   end
 
+  # Retrieve records from the Web of Science
+  # Web of Science results are based on partial string matching for DOI queries, unfortunately.
+  # @see WebOfScience::Queries#search_by_doi
+  # @param [String] doi
+  # @return [Array<Hash>] matching hashes
+  def self.web_of_science(doi)
+    return [] unless Settings.WOS.enabled
+    doi = doi_name(doi)
+    return [] if doi.blank?
+    retriever = WebOfScience.queries.search_by_doi(doi)
+    retriever.next_batch.map { |record| add_citation(record.pub_hash) }
```

Comment:
All this, and we **still** don't get an Enumerator over all the records regardless of what `batch` they are in?  Why is this considered an improvement?